src/mkpsxiso/cdwriter.cpp

```diff
@@ -101,8 +102,10 @@ void IsoWriter::SectorView::CalculateForm1()
 void IsoWriter::SectorView::CalculateForm2()
 {
 	SECTOR_M2F2* sector = static_cast<SECTOR_M2F2*>(m_currentSector);
-	m_checksumJobs.emplace_front(std::async(std::launch::async, &EDCECC::ComputeEdcBlock, &EDC_ECC_GEN,
-		sector->data, sizeof(sector->data) - 4, sector->data + sizeof(sector->data) - 4));
+	m_checksumJobs.emplace_front(m_threadPool.enqueue([](SECTOR_M2F2* sector)
+	{
+		EDC_ECC_GEN.ComputeEdcBlock(sector->data, sizeof(sector->data) - 4, sector->data + sizeof(sector->data) - 4);
+	}, sector));

```

Comment:
Looking at https://github.com/progschj/ThreadPool/blob/master/ThreadPool.h#L18-L19, I think this change may not have been needed. Just replacing `std::async(std::launch::async` with `m_threadPool.enqueue` might be enough, as those arguments get forwarded to `std::bind` that has the same semantics as async's constructor.

Reply:
I remember trying to not change that part, for the same reasons you suggest, but I was getting compile errors, otherwise.

Reply:
In any case, I was finding this version more explicit on what is the task being performed.

Reply:
In any case, if you prefer the previous form, adding a & before EDCECC::ComputeEdcBlock should be enough.

Reply:
Yes, just like it was before this change.

Reply:
It should now be like before, but using the thread pool.